problems/add_two_numbers.py

```diff
@@ -0,0 +1,41 @@
+#https://leetcode.com/problems/add-two-numbers/submissions/
+
+# Definition for singly-linked list.
+# class ListNode(object):
+#     def __init__(self, val=0, next=None):
+#         self.val = val
+#         self.next = next
+
+class Solution(object):
+    def addTwoNumbers(self, l1, l2):
+        """
+        :type l1: ListNode
+        :type l2: ListNode
+        :rtype: ListNode
+        """
+        l3 = ListNode()
+        cur1 = l1
+        cur2 = l2
+        cur3 = l3
```

Comment:
Using a third linked list is smart to preserve the data of the input. worth pointing out in an interview that you're trading off O(1) space to get a more predictable program